src/assembler.rs

```diff
@@ -0,0 +1,361 @@
+use std::collections::HashSet;
+
+use crate::{
+    formatter::append_and_tab,
+    structures::*,
+};
+use convert_case::{
+    Case,
+    Casing,
+};
+
+/// Assembles ink! contract from the parsed contract struct and return it as a vec of Strings
+pub fn assemble_contract(contract: Contract) -> Vec<String> {
+    let mut output_vec = Vec::<String>::new();
```

Comment:
Did you try to find some library with templates or something like this? Because I'm okay with adding it line by line, but maybe you can have a more readable pattern/template as we used in typechain.
```
#![cfg_attr(not(feature = "std"), no_std)]
#![feature(min_specialization)]

#[brush::contract]
pub mod ${ name_of_contract } {
...
}
```

Reply:
Maybe `quote` and `syn` libraries can be useful for you=)